crates/ruff/src/rules/pylint/rules/bad_super_call.rs

```diff
@@ -0,0 +1,103 @@
+
+#[violation]
+pub struct BadSuperCall {
+    base: String,
+    class: String,
+}
+
+impl Violation for BadSuperCall {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let BadSuperCall { base, class } = self;
+        format!("Bad Super Call `{base}` is not super for class `{class}`")
+    }
+}
+/// Returns `true` if a call is an argumented `super` invocation.
+fn is_super_call_with_arguments(func: &Expr, args: &[Expr]) -> bool {
+    if let Expr::Name(ast::ExprName { id, .. }) = func {
+        id == "super" && !args.is_empty()
+    } else {
+        false
+    }
+}
+
+/// PLE0241
+/// pub(crate) fn bad_super_call(checker: &mut Checker, name: &str, bases: &[Expr]) {
+pub(crate) fn bad_super_call(
+    checker: &mut Checker,
+    expr: &Expr,
+    func: &Expr,
+    args: &[Expr],
+) {
+    // Only bother going through the super check at all if we're in a `super` call.
+    // (We check this in `super_args` too, so this is just an optimization.)
+    if !is_super_call_with_arguments(func, args) {
+        return;
+    }
+    let scope = checker.ctx.scope();
+
+    // Check: are we in a Function scope?
+    if !matches!(scope.kind, ScopeKind::Function(_)) {
+        return;
+    }
+
+    let mut parents = checker.ctx.parents();
+
+    // For a `super` invocation to be unnecessary, the first argument needs to match
+    // the enclosing class, and the second argument needs to match the first
+    // argument to the enclosing function.
+    let [first_arg, second_arg] = args else {
+        return;
+    };
+    // Find the enclosing function definition (if any).
+    let Some(Stmt::FunctionDef(ast::StmtFunctionDef {
+        args: parent_args, ..
+    })) = parents
+        .find(|stmt| stmt.is_function_def_stmt()) else {
+        return;
+    };
+
+    // Extract the name of the first argument to the enclosing function.
+    let Some(Arg {
+        arg: parent_arg, ..
+    }) = parent_args.args.first() else {
+        return;
+    };
+    let Expr::Name(ast::ExprName {id: first_arg_id, ..}) = first_arg else {
+        return;
+    };
+
+    // Find the enclosing class definition (if any).
+    let Some(Stmt::ClassDef(ast::StmtClassDef {
+        name: parent_name, ..
+    })) = parents
+        .find(|stmt| matches!(stmt, Stmt::ClassDef (_))) else {
+        return;
+    };
+
+    // Need help here, do not know how to get the base_class name
+    let base_class_stmt = checker.ctx.stmt(); // ??? wrong 
+    // How do I get  the Identifier of the base class?
+    let Some(Stmt::ClassDef(ast::StmtClassDef {
+        name: base_class_id, ..
+    })) = base_class_stmt else {
+        return;
+    };
+
+    if first_arg_id != parent_name || first_arg_id != base_class_id  {
+        checker.diagnostics.push(Diagnostic::new(
+            BadSuperCall{
+                base: base_class_id.to_string(),
+                class: parent_name.to_string(),
+            },
+            expr.range(),
+        ));
+    }
```

Comment:
A fixer will be implemented after I know how to get the base class

Reply:
Regarding a fix, there are a few ways we can go about it.

Example:
```python

class Animal:
    pass


class Tree:
    pass


class Cat(Animal):
    def __init__(self):
        super(Tree, self).__init__()
```

The super call could be to the correct class the user wants and the error is in the inheritance class.

So `Animal` could be the wrong class and the fix should instead make `Tree` the parent class.

So for this rule, either `Animal` or `Tree` could be the problem, depending on what the user wants to do. Imo that means this violation cannot be auto-fixable, unless we look at context (but I'd rather the user be the one looking at context).

I prefer violations that are auto-fixable, so looking for opinions if I should just implement something that changes `Tree` to `Animal` in the super call, or not do anything for the fixing.

Reply:
I think it's too difficult to know what the "correct" behavior is here. It's also possible that neither `Tree` nor `Animal` are the intended argument to `super` -- the user could be trying to invoke `super` on some other parent class in the chain. My advice would be to avoid implementing a fix.